### PR TITLE
MAT-6651 Crosswalk QI-Core type over to correct FHIR type for dataRequirement.type correctly

### PR DESCRIPTION
**Pull request by adongare**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-6651](https://jira.cms.gov/browse/MAT-6651)
(Optional) Related Tickets:

### Summary
- refactored data requirement module definition library generation
- introduced a recursive flag
- related PR: https://github.com/MeasureAuthoringTool/madie-fhir-service/pull/193

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/test/java/gov/cms/mat/cql_elm_translation/service/CqlConversionServiceTest.java [hunk 1]
```diff
@@ -182,6 +182,9 @@ void testProcessCqlDataWithErrorsMissingModel() {
   @Test
   void testGetTranslatedLibrariesForCqlForCql() throws IOException {
     String cql = getData("/qdm_data_criteria_retrieval_test.cql");
+    MadieLibrarySourceProvider.setUsing(new CqlTextParser(cql).getUsing());
+    MadieLibrarySourceProvider.setCqlLibraryService(cqlLibraryService);
+    MadieLibrarySourceProvider.setAccessToken("access token");
```
Comment (adongare) [inline src/test/java/gov/cms/mat/cql_elm_translation/service/CqlConversionServiceTest.java:187]: this is unrelated to this ticket. I will revert if anyone objects. this is to fix the gundam build failure: https://qnetjenkins.cms.gov/bmat-master/job/Madie/job/Madie%20Services/job/cql-elm-translator-service/job/sprint%252Fgundam_2024-01-29/3/console.

File: src/test/java/gov/cms/mat/cql_elm_translation/service/CqlConversionServiceTest.java [hunk 2]
```diff
@@ -182,11 +185,20 @@ void testProcessCqlDataWithErrorsMissingModel() {
...
     List<TranslatedLibrary> libraries = service.getTranslatedLibrariesForCql(cql, "token");
     AtomicBoolean foundAMatch = new AtomicBoolean();
-    libraries.forEach(
-        library -> foundAMatch.set(library.getElmJson().contains("DataCriteriaRetrivalTest")));
-    assertThat(foundAMatch.get(), is(true));
+    var matchingLib =
+        libraries.stream()
+            .filter(library -> library.getElmJson().contains("DataCriteriaRetrivalTest"))
+            .findFirst();
+    assertThat(matchingLib.get().getName(), is(equalTo("DataCriteriaRetrivalTest")));
```
Comment (adongare) [inline src/test/java/gov/cms/mat/cql_elm_translation/service/CqlConversionServiceTest.java:201]: unrelated to this ticket. I will revert if anyone objects. this is to fix the gundam build failure: https://qnetjenkins.cms.gov/bmat-master/job/Madie/job/Madie%20Services/job/cql-elm-translator-service/job/sprint%252Fgundam_2024-01-29/3/console.

**Review by RohitKandimalla — APPROVED** [review]

File: src/main/java/gov/cms/mat/cql_elm_translation/service/EffectiveDataRequirementService.java [hunk 3]
```diff
@@ -61,43 +36,19 @@ private RequestData createDefaultRequestData(String identifier, String cql) {
...
-                    "Measure", entry.getResource().getResourceType().toString()))
-        .findFirst();
-  }
-
-  /**
-   * @param r5Measure retrieved from measure bundle
-   * @param library measure library
-   * @param accessToken used by MadieLibrarySourceProvider to make calls to madie-fhir-services
+   * @param libraryDetails - instance of CqlLibraryDetails
+   * @param recursive- indicates if the data requirements gathered should be recursive
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/mat/cql_elm_translation/service/EffectiveDataRequirementService.java:40]: Might be helpful to future us to leave a note of what "recursive" means in this context.

**Review by jkotanchik-SB — APPROVED** [review]

File: src/main/java/gov/cms/mat/cql_elm_translation/service/EffectiveDataRequirementService.java (hunk 3 above)
Comment (adongare) [reply src/main/java/gov/cms/mat/cql_elm_translation/service/EffectiveDataRequirementService.java:40]: my understanding of `recursive` does not match with what this flag is doing. I'm going to schedule a call with Bryn to learn more about it. Accordingly, we'll either update it or revert it and find another fix.
